File: c_00/ex03/ft_print_numbers.c
/* ************************************************************************** */
/*                                                                            */
/*                                                        :::      ::::::::   */
/*   ft_print_numbers.c                                 :+:      :+:    :+:   */
/*                                                    +:+ +:+         +:+     */
/*   By: photon <marvin@42.fr>                      +#+  +:+       +#+        */
/*                                                +#+#+#+#+#+   +#+           */
/*   Created: 2023/06/21 16:36:20 by photon            #+#    #+#             */
/*   Updated: 2023/06/21 17:09:18 by photon           ###   ########.fr       */
/*                                                                            */
/* ************************************************************************** */
#include <unistd.h>

void	ft_print_numbers(void)
{
	char	num_disp;

	num_disp = '0';
	while (num_disp <= '9')
	{
		write(1, &num_disp, 1);
		num_disp++;
	}
}
/*
int	main(void)
{
	ft_print_numbers();
	return (0);
}
*/

File: zig/zig_00/ex03/ft_print_numbers.zig
const std = @import("std");

pub fn ft_putchar(c: u8) void {
    const buf: [1]u8 = [_]u8{c};
    _ = std.os.write(1, &buf) catch return;
}

pub fn ft_print_numbers() void {
    var num: u8 = '0';
    while (num <= '9')
    {
        ft_putchar(num);
        num += 1;
    }
}

pub fn main() !void {
    ft_print_numbers();
}

